File: guests/python/panes.py
"""The panes scene (tools/scenes/panes.steps): nothing here is
panes-specific except `panes=3`, asked for ONCE."""

import sys

import kaya

app = kaya.App()

CONTENT = 7
DETAIL = 8


def open_detail():
    with app.push_entry(DETAIL, title="detail"):
        caption = kaya.signal("detail pane")
        with kaya.column():
            kaya.label(bind=caption).a11y_id("detail")  # label#last


def open_content():
    with app.push_entry(CONTENT, title="content"):
        caption = kaya.signal("content pane")
        with kaya.column():
            kaya.label(bind=caption).a11y_id("content")  # label#1
            kaya.button("open detail", on_click=open_detail)  # button#1


with app.window(title="panes", panes=3):
    caption = kaya.signal("root pane")
    with kaya.column():
        # Authored ids: an index read passes for an empty arm.
        kaya.label(bind=caption).a11y_id("root")  # label#0
        kaya.button("open content", on_click=open_content)  # button#0


sys.exit(app.run())

File: guests/js/panes.ts
// The panes scene (tools/scenes/panes.steps): nothing here is
// panes-specific except `panes: 3`, asked for ONCE.

import * as kaya from "kaya-gui";

const app = new kaya.App();

const CONTENT = 7;
const DETAIL = 8;

function openDetail(): void {
  app.pushEntry(DETAIL, { title: "detail" }, () => {
    const caption = kaya.signal("detail pane");
    kaya.column(() => {
      kaya.label({ bind: caption }).a11yId("detail"); // label#last
    });
  });
}

function openContent(): void {
  app.pushEntry(CONTENT, { title: "content" }, () => {
    const caption = kaya.signal("content pane");
    kaya.column(() => {
      kaya.label({ bind: caption }).a11yId("content"); // label#1
      kaya.button("open detail", { onClick: openDetail }); // button#1
    });
  });
}

app.window({ title: "panes", panes: 3 }, () => {
  const caption = kaya.signal("root pane");
  kaya.column(() => {
    // Authored ids: an index read passes for an empty arm.
    kaya.label({ bind: caption }).a11yId("root"); // label#0
    kaya.button("open content", { onClick: openContent }); // button#0
  });
});

app.run();
